tools/http/src/http_client.rs

```diff
@@ -0,0 +1,500 @@
+    }
+
+    /// Builds request body
+    pub fn build_body(
+        &self,
+        request: reqwest::RequestBuilder,
+        body: &RequestBody,
+        method: &reqwest::Method,
+    ) -> Result<reqwest::RequestBuilder, HttpToolError> {
+        // Skip body for methods that don't support it
+        if !Self::method_supports_body(method) {
+            return Ok(request);
+        }
+        match body {
+            RequestBody::Json { data } => Ok(request.json(data)),
+            RequestBody::Form { data } => Ok(request.form(data)),
+            RequestBody::Multipart { fields } => self.build_multipart_form(request, fields),
+            RequestBody::Raw { data, content_type } => {
+                self.build_raw_body(request, data, content_type)
+            }
+        }
+    }
+
+    /// Builds multipart form from fields
+    fn build_multipart_form(
+        &self,
+        request: reqwest::RequestBuilder,
+        fields: &[crate::models::MultipartField],
+    ) -> Result<reqwest::RequestBuilder, HttpToolError> {
+        let mut form = Form::new();
+        for field in fields {
+            let mut part = reqwest::multipart::Part::text(field.value.clone());
+
+            // Set content type if provided
+            if let Some(content_type) = &field.content_type {
+                part = part
+                    .mime_str(content_type)
+                    .map_err(|e| HttpToolError::ErrInput(format!("Invalid content type: {}", e)))?;
+            }
+
+            form = form.part(field.name.clone(), part);
+        }
+        Ok(request.multipart(form))
+    }
+
+    /// Builds raw body from base64 data
+    fn build_raw_body(
+        &self,
+        request: reqwest::RequestBuilder,
+        data: &str,
+        content_type: &Option<String>,
+    ) -> Result<reqwest::RequestBuilder, HttpToolError> {
+        let bytes = base64::engine::general_purpose::STANDARD
+            .decode(data)
+            .map_err(|e| HttpToolError::ErrBase64Decode(format!("Invalid base64 data: {}", e)))?;
+
+        let content_type = content_type
+            .as_deref()
+            .unwrap_or("application/octet-stream");
+
+        Ok(request.header("Content-Type", content_type).body(bytes))
+    }
+
+    /// Executes the request and returns the response
+    pub async fn execute(
+        &self,
+        request: reqwest::RequestBuilder,
+    ) -> Result<reqwest::Response, HttpToolError> {
+        request
+            .send()
+            .await
+            .map_err(HttpToolError::from_network_error)
+    }
+
+    /// Executes a request with retry logic
+    pub async fn execute_with_retry(
+        &self,
+        request: reqwest::RequestBuilder,
+        retries: u32,
+    ) -> Result<reqwest::Response, HttpToolError> {
+        if retries == 0 {
+            // No retries needed, execute once
+            return self.execute(request).await;
+        }
+
+        let mut last_error = None;
+
+        for attempt in 0..=retries {
```

Comment:
Instead of writing our own retry logic, let's simplify the code using [`backon`](https://docs.rs/backon/latest/backon/) for it.